AudioKitSynthOne/DSP/Kernel/S1DSPKernel+startStopNotes.mm

```diff
@@ -54,8 +67,19 @@ static inline double tuningTableNoteToHz(int noteNumber) {
     if (noteNumber < 0 || noteNumber >= S1_NUM_MIDI_NOTES)
         return;
 
-    NSNumber* nn = @(noteNumber);
-    [heldNoteNumbers removeObject: nn];
+    NSInteger index = -1;
+    NoteNumber existingNote;
+    for(int i = 0; i < heldNoteNumbers.count; i++) {
+        NSValue* value = heldNoteNumbers[i];
+        [value getValue:&existingNote];
+        if(existingNote.noteNumber == noteNumber) {
+            index = i;
+            break;
+        }
+    }
+    if(index != -1)
+        [heldNoteNumbers removeObjectAtIndex:index];
```

Comment:
> Left a few notes, other than that looks good. Though I dont know what the [getValue] call does in the realtime thread heldnotesarray now (since its an objc call?)

This goes back to your idea of replacing Tyson's "AE" code.

The intention is that the objc heldnotesarray is never accessed from the realtime thread.  *heldNotesArrayAE is*.  

If you see heldnotesarray being accessed from the realtime thread that is a bug and we should figure out how to fix it.

Reply:
Looks good to me.

Reply:
> Since you call `break;` in the loop anyway, what speaks against removing the object and the index there? (might get rid off one extra if-condition)

I fixed this but I need to commit it and update the PR (or just merge it cause it's trivial and I tested it)